src/main/java/uk/nhs/hee/tis/revalidation/connection/service/ExceptionService.java

```diff
@@ -72,42 +67,23 @@ public void removeExceptionLog(final String gmcId) {
   }
 
   /**
-   * Get exception log.
+   * Get today's exception logs for a specific admin
    *
-   * @param requestDto request for getting exception log
-   */
-  public ExceptionResponseDto getExceptionLog(final ExceptionRequestDto requestDto) {
-    final var direction = "asc".equalsIgnoreCase(requestDto.getSortOrder()) ? ASC : DESC;
-    final var pageableAndSortable = of(requestDto.getPageNumber(), 20,
-        by(direction, requestDto.getSortColumn()));
-
-    final var exceptionLogPage = repository.findAll(pageableAndSortable);
-    final var exceptionLogs = exceptionLogPage.get().collect(toList());
-    return ExceptionResponseDto.builder()
-        .totalPages(exceptionLogPage.getTotalPages())
-        .totalResults(exceptionLogPage.getTotalElements())
-        .exceptionRecord(buildExceptionRecords(exceptionLogs))
-        .build();
-  }
-
-  /**
-   * Get exception log hashmap.
+   * @param admin request by date today
    */
-  public Map<String, String> getExceptionsMap() {
-    Map<String,String> exceptionsMap = new HashMap<>();
-    List<ExceptionLog> exceptionLogList = repository.findAll();
-    for (ExceptionLog exceptionLog: exceptionLogList) {
-      exceptionsMap.put(exceptionLog.getGmcId(), exceptionLog.getErrorMessage());
-    }
-    return exceptionsMap;
-  }
+  public List<ExceptionLogDto> getExceptionLogs(String admin) {
+    LocalDateTime today = LocalDate.now().atStartOfDay();
+    LocalDateTime tomorrow = today.plusDays(1);
 
-  private List<ExceptionRecordDto> buildExceptionRecords(final List<ExceptionLog> exceptionLogs) {
-    return exceptionLogs.stream().map(exception -> {
-      return ExceptionRecordDto.builder()
-          .gmcId(exception.getGmcId())
-          .exceptionMessage(exception.getErrorMessage())
+    final var todaysExceptions = repository.findByAdminAndTimestampBetween(admin, today, tomorrow);
+    return todaysExceptions.stream().map(exceptionLog -> {
+      return ExceptionLogDto.builder()
```

Comment:
Could we have a mapper to map ExceptionLog -> ExceptionLogDto?

Reply:
ok sure. Here it is @hihilary